include/votca/xtp/pmdecomposition.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+#ifndef VOTCA_XTP_PMDECOMPOSITION_H
+#define VOTCA_XTP_PMDECOMPOSITION_H
+#include "logger.h"
+#include "votca/xtp/orbitals.h"
+
+namespace votca {
+namespace xtp {
+
+class PMDecomposition {
+ public:
+  PMDecomposition(Logger &log_) : log(log_){};
+  void computePMD(Orbitals &orbitals);
+
+ private:
+  Logger &log;
+  Eigen::MatrixXd rotatedorbitals(Eigen::MatrixXd &maxorbs, Index s, Index t);
+  Eigen::MatrixXd orbitalselections(Eigen::MatrixXd &m,
+                                    const Eigen::MatrixXd &S);
+  void update_maximums(Eigen::MatrixXd &m, Index col1, Index col2,
+                       Eigen::MatrixXd &new_orbs);
+  BasisSet basis;
+  AOBasis aobasis;
+  Eigen::MatrixXd A;
+  Eigen::MatrixXd B;
```

Comment:
I do not think you need any of those. If so please make names A and B more descriptive